src/systems/systems/src/pressurization/cabin_pressure_controller.rs

```diff
@@ -0,0 +1,475 @@
+        // Convert cabin V/S to pressure/delta
+        const KPA_FT: f64 = 0.0366; //ASL
+
+        let cab_press_var =
+            -self.cabin_vs.get::<foot_per_minute>() * KPA_FT / 60. * context.delta().as_secs_f64();
+        self.cabin_pressure =
+            self.last_cabin_pressure + Pressure::new::<hectopascal>(cab_press_var);
+    }
+
+    fn update_cabin_altitude(&mut self, sea_level_pressure: Pressure, destination_qnh: Pressure) {
+        // ISA constants for calculating cabin altitude
+        const T_0: f64 = 288.2; // K
+        const R: f64 = 287.058; // m2/s2/K
+        const L: f64 = -0.00651; // K/m
+        const G: f64 = 9.80665; // m/s2
+
+        let p_0: f64;
+        let p: f64 = self.cabin_pressure().get::<hectopascal>();
+
+        // Based on local QNH when below 5000ft from departure or arrival airport, ISA when above
+        if self.pressure_schedule() == PressureSchedule::DescentInternal
+            && (self.cabin_altitude() - self.landing_elevation())
+                .get::<foot>()
+                .abs()
+                < 5000.
+        {
+            if destination_qnh > Pressure::new::<hectopascal>(0.) {
+                p_0 = destination_qnh.get::<hectopascal>();
+            } else {
+                p_0 = sea_level_pressure.get::<hectopascal>();
+            }
+        } else if (self.cabin_altitude() - self.departure_elevation())
+            .get::<foot>()
+            .abs()
+            < 5000.
+        {
+            p_0 = sea_level_pressure.get::<hectopascal>();
+        } else {
+            p_0 = 1013.25;
+        }
+
+        // Vars
+        let pressure_ratio: f64 = p / p_0;
+
+        // Hydrostatic equation with linear temp changes and constant R, g
+        let z: f64 = ((T_0 / pressure_ratio.powf((L * R) / G)) - T_0) / L;
+        self.cabin_alt = Length::new::<meter>(z);
+    }
+
+    fn update_cabin_vs(&mut self, context: &UpdateContext) {
+        match self.pressure_schedule() {
+            PressureSchedule::Ground => {
+                if context.is_on_ground() {
+                    self.cabin_pressure = self.exterior_pressure;
+                } else {
+                    // Formula to simulate pressure start state if starting in flight
+                    let ambient_pressure: f64 = context.ambient_pressure().get::<hectopascal>();
+                    self.cabin_pressure = Pressure::new::<hectopascal>(
+                        -0.0002 * ambient_pressure.powf(2.) + 0.5463 * ambient_pressure + 658.85,
+                    );
+                }
+                self.cabin_target_vs = Velocity::new::<foot_per_minute>(0.);
+            }
+            PressureSchedule::TakeOff => {
+                if self.cabin_delta_p() < Pressure::new::<psi>(0.1) {
+                    self.cabin_target_vs = Velocity::new::<foot_per_minute>(-400.)
+                } else {
+                    self.cabin_target_vs = Velocity::new::<foot_per_minute>(0.)
+                }
+            }
```

Comment:
As discussed on Discord, I've left this as is because I couldn't find an optimal solution to move this code to the `PressureSchedule` `enum` `impl` block. Happy to continue discussing as I probably just don't have enough knowledge of `enum`'s to make this work! 😄